File: app/src/main/java/nu/huw/clarity/ClarityApplication.java
package nu.huw.clarity;

import android.app.Application;
import com.jakewharton.threetenabp.AndroidThreeTen;

public class ClarityApplication extends Application {

  @Override
  public void onCreate() {
    super.onCreate();
    AndroidThreeTen.init(this);
  }
}

File: app/src/main/java/nu/huw/clarity/ClarityApplication.kt
package nu.huw.clarity

import android.app.Application
import com.jakewharton.threetenabp.AndroidThreeTen
import nu.huw.clarity.dagger.AppComponent
import nu.huw.clarity.dagger.AppModule
import nu.huw.clarity.dagger.DaggerAppComponent
import nu.huw.clarity.db.AppDatabase
import nu.huw.clarity.model.Task
import javax.inject.Inject

class ClarityApplication : Application() {

    @Inject lateinit var db: AppDatabase

    val component: AppComponent by lazy {
        DaggerAppComponent.builder().appModule(AppModule(this)).build()
    }

    override fun onCreate() {
        super.onCreate()
        component.inject(this)
        AndroidThreeTen.init(this)
    }
}
